src/Altinn.Profile.Integrations/OrganizationNotificationAddress/OrganizationNotificationAddressUpdateJob.cs

```diff
@@ -0,0 +1,65 @@
+/// </summary>
+/// <param name="organizationNotificationAddressSettings">Settings for the synchronization update job</param>
+/// <param name="organizationNotificationAddressHttpClient">A HTTP client that can be used to retrieve contact details changes</param>
+/// <param name="metadataRepository">A repository implementation for managing persistence of the job status between runs</param>
+/// <param name="notificationAddressUpdater">A repository implementation for managing persistence for the local contact information</param>
+public class OrganizationNotificationAddressUpdateJob(
+    OrganizationNotificationAddressSettings organizationNotificationAddressSettings,
+    IOrganizationNotificationAddressHttpClient organizationNotificationAddressHttpClient,
+    IRegistrySyncMetadataRepository metadataRepository,
+    IOrganizationNotificationAddressUpdater notificationAddressUpdater)
+    : IOrganizationNotificationAddressUpdateJob
+{
+    private readonly OrganizationNotificationAddressSettings _organizationNotificationAddressSettings = organizationNotificationAddressSettings;
+    private readonly IOrganizationNotificationAddressHttpClient _organizationNotificationAddressHttpClient = organizationNotificationAddressHttpClient;
+    private readonly IRegistrySyncMetadataRepository _metadataRepository = metadataRepository;
+    private readonly IOrganizationNotificationAddressUpdater _notificationAddressUpdater = notificationAddressUpdater;
+
+    /// <inheritdoc/>
+    /// <exception cref="InvalidOperationException">Thrown when the endpoint URL is null or empty.</exception>
+    public async Task SyncNotificationAddressesAsync()
+    {
+        if (string.IsNullOrWhiteSpace(_organizationNotificationAddressSettings.ChangesLogEndpoint))
+        {
+            throw new InvalidOperationException("The endpoint URL must not be null or empty.");
+        }
+
+        DateTime lastUpdated = await _metadataRepository.GetLatestSyncTimestampAsync();
+
+        // Time should be in iso8601 format example: 2018-02-15T11:07:12Z
+        string? fullUrl = _organizationNotificationAddressSettings.ChangesLogEndpoint + $"?since={lastUpdated.ToString("yyyy-MM-ddTHH\\:mm\\:ss.fffZ")}&pageSize={_organizationNotificationAddressSettings.ChangesLogPageSize}";
+
+        do
+        {
+            NotificationAddressChangesLog changesLog = await _organizationNotificationAddressHttpClient.GetAddressChangesAsync(fullUrl);
+
+            var noChangesSinceLastCheck = changesLog.OrganizationNotificationAddressList == null || changesLog.OrganizationNotificationAddressList?.Count == 0;
+            if (noChangesSinceLastCheck)
+            {
+                break;
+            }
+
+            int updatedRowsCount = await _notificationAddressUpdater.SyncNotificationAddressesAsync(changesLog);
+
+            if (updatedRowsCount > 0 && changesLog.Updated.HasValue)
+            {
+                var lastUpdatedTimestamp = changesLog.Updated;
```

Comment:
I'm a bit unsure about this.
As the code is written now, this timestamp will be updated for every page in the changelog. The updated timestamp will reflect the time of getting the data. 
 Normally if this is run every 10 minutes I think we only will have 1 page of data and thus there is no problem with this code. 
 For the first sync batch, we will have many pages, and this code will update the sync time in the DB for every page - but this will be the same timestamp. 
 
 Imagine a case where we start syncing and everything goes fine until a date in ex 2022, when data is corrupted. We will still update the timestamp with the current timestamp. Then the next shceduled run will only get data from the last timestamp and we will possibly need to update the db manually to get data between 2022 and 2025. 
 An alternative is to update the latestChangeTimestamp to the last entry's "updated"-timestamp. The caveat to this is that it does not feel like it's how the api is designed to work.
 
 @SandGrainOne @hggutvik 
```suggestion
                var lastUpdatedTimestamp = changesLog.OrganizationNotificationAddressList.Last().Updated;
```

Reply:
Nice find. I think we need to use the timestamp of the last change on the page. That feels like the safer option. This would also mean no update of metadata if there are no changes.